### DISCUSSION TITLE
how to use ecapture with android emulator

### DISCUSSION THREAD

**Discussion by ghost** [gojue/ecapture#350] [General]
I want to use ecapture with android emualtor. Say (system-images\;android-33\;google_apis\;x86_64). I try to download https://github.com/gojue/ecapture/releases/download/v0.5.1/ecapture-v0.5.1-android-x86_64.tar.gz and extract ecapture and adb push ecapture /data/local/tmp
Then I adb root and adb shell
I try
emu64xa:/ # /data/local/tmp/ecapture tls -w save_android.pcapng -i wlan0 --libssl=/apex/com.android.conscrypt/lib64/libssl.so --ssl_version="boringssl 1.1.1" --port 443                                                                     
tls_2023/04/20 03:12:15 ECAPTURE :: ecapture Version : androidgki_x86_64:0.5.1-20230408-e1afbb8:5.15.0-1035-azure
tls_2023/04/20 03:12:15 ECAPTURE :: Pid Info : 5750
tls_2023/04/20 03:12:15 ECAPTURE :: Kernel Info : 5.15.41
tls_2023/04/20 03:12:15 EBPFProbeOPENSSL	module initialization
tls_2023/04/20 03:12:15 EBPFProbeOPENSSL	module initialization failed, [skip it]. error:open ecapture_masterkey.log: read-only file system
tls_2023/04/20 03:12:15 ECAPTURE :: 	No runnable modules, Exit(1)
Any idea about how to make it work?

**Comment by cfc4n** [comment]
"read-only file system"?

**Reply by ghost** [reply]
Yes, this is the output of ecapture. Could you reproduce in your side?

I use the following command to run the emulator on linux mint 21.1:
curl https://dl.google.com/android/repository/commandlinetools-linux-9123335_latest.zip -o commandline.zip
unzip commandline.zip
rm -rf commandline.zip
mv cmdline-tools latest
mkdir -p sdk/cmdline-tools
mv latest sdk/cmdline-tools
echo y | sdk/cmdline-tools/latest/bin/sdkmanager --no_https system-images\;android-30\;google_apis\;x86_64 platform-tools platforms\;android-33 --channel=0
echo no | sdk/cmdline-tools/latest/bin/avdmanager create avd -f -n android -k system-images\;android-30\;google_apis\;x86_64
awk -i inplace {sub\(/320/\,\"600\"\)\;sub\(/640/\,\"900\"\)}1 ~/.android/avd/android.avd/config.ini
sdk/emulator/emulator -avd android -wipe-data -no-snapshot -no-audio -no-boot-anim -writable-system -memory 4096 -gpu swiftshader_indirect &
adb wait-for-device
adb root
while [[ $(adb exec-out getprop sys.boot_completed) != 1 ]]
do
    sleep 30
done
adb shell avbctl disable-verification
adb reboot
adb wait-for-device
adb root
while [[ $(adb exec-out getprop sys.boot_completed) != 1 ]]
do
    sleep 30
done
adb exec-out getprop sys.boot_completed
adb remount
adb devices -l

But still suffer from the read-only file system problem. Please help.

**Comment by cfc4n** [comment]
The environment I am using is `android Studio` and with image `Pixel_3a_API_33_x86_64`.

<img width="855" alt="image" src="https://user-images.githubusercontent.com/709947/233665273-fd30f4b6-c769-40d0-b996-de78156dc340.png">
